src/Definition.php

```diff
@@ -41,7 +41,14 @@ final class Definition
     private $maker;
 
     /**
-     * The callback.
+     * The stack of callbacks.
+     *
+     * @var callable[]
+     */
+    private $callbackStack = [];
```

Comment:
Why do we need this as well as a callback variable. Can we remove the callback property?

Reply:
We don't need both, I used the callback property to avoid recreating the lambda on every getCallback(). That may be premature optimization though.

Reply:
I've removed the callback field and just used callbackStack directly in getCallback()